security/identity-and-access-management/oracle-identity-governance/event-handler-samples/user-lifecycle-event-notification/files/src/com/oracle/sample/UserLifecycleEventNotification.java

```diff
@@ -0,0 +1,372 @@
+ * SOFTWARE.
+ *******************************************************************************/
+
+package com.oracle.sample;
+
+import java.io.Serializable;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import com.thortech.xl.dataaccess.tcDataProvider;
+import com.thortech.xl.dataobj.util.XLDatabase;
+import com.thortech.xl.dataobj.util.tcEmailNotificationUtil;
+
+import Thor.API.tcResultSet;
+import Thor.API.Exceptions.tcAPIException;
+import Thor.API.Exceptions.tcColumnNotFoundException;
+import Thor.API.Exceptions.tcInvalidLookupException;
+import Thor.API.Operations.tcLookupOperationsIntf;
+import oracle.iam.identity.exception.UserSearchException;
+import oracle.iam.identity.usermgmt.api.UserManager;
+import oracle.iam.identity.usermgmt.api.UserManagerConstants;
+import oracle.iam.identity.usermgmt.vo.User;
+import oracle.iam.platform.Platform;
+import oracle.iam.platform.authz.exception.AccessDeniedException;
+import oracle.iam.platform.context.ContextAware;
+import oracle.iam.platform.entitymgr.vo.SearchCriteria;
+import oracle.iam.platform.kernel.OrchestrationEngine;
+import oracle.iam.platform.kernel.spi.PostProcessHandler;
+import oracle.iam.platform.kernel.vo.AbstractGenericOrchestration;
+import oracle.iam.platform.kernel.vo.BulkEventResult;
+import oracle.iam.platform.kernel.vo.BulkOrchestration;
+import oracle.iam.platform.kernel.vo.EventResult;
+import oracle.iam.platform.kernel.vo.Orchestration;
+
+public class UserLifecycleEventNotification implements PostProcessHandler {
+	private static final Logger logger = Logger.getLogger(UserLifecycleEventNotification.class.getName());
+
+	private static final String ORCHESTRATION_CREATE_EVENT = "CREATE";
+
+	private static final String LOOKUP_NAME = "Lookup.Sample.LifecycleNotification";
+	private static final String LOOKUP_RECIPIENT_EMAILS = "Recipient Emails";
+	private static final String LOOKUP_RECIPIENT_SELF = "Self Recipient";
+	private static final String LOOKUP_USER_STATE = "User State";
+	private static final String LOOKUP_TEMPLATE_NAME = "Template Name";
+
+	private static final String LOOKUP_FIELD_CODE = "Lookup Definition.Lookup Code Information.Code Key";
+	private static final String LOOKUP_FIELD_DECODE = "Lookup Definition.Lookup Code Information.Decode";
+
+	private static final String TEMPLATE_SELECT_QUERY = "SELECT NOTIFICATIONTEMPLATE.ID, LONGMESSAGE, SUBJECT "
+			+ "FROM LOCALTEMPLATE, NOTIFICATIONTEMPLATE WHERE LOCALTEMPLATE.TEMPLATEID = NOTIFICATIONTEMPLATE.ID AND TEMPLATENAME = ?";
+
+	private static final String FROM_EMAIL_ADDRESS = "donotreply@example.com";
+
+	@Override
+	public EventResult execute(long arg0, long arg1, Orchestration arg2) {
+		logger.log(Level.INFO, "Event handler execution has started.");
```

Comment:
Use logger.info() instead of logger.log(Level.INFO).
Applies to all usages of logger.log(), i.e., use the convenience method instead of specifying the level.